### PR TITLE
SCRUM-3062 Load Biological Spatial Ontology (BSPO)

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java [hunk 1]
```diff
@@ -239,7 +239,7 @@ private boolean isNodeInOntology(OWLClass currentTreeNode, T currentTerm, HashSe
 		boolean condition5 = !config.getIgnoreEntitiesWithChebiXref();
 		boolean condition6 = !hasChebiXref(currentTerm);
 
-		return ((condition1 && condition2) || (condition3 && condition4)) && (condition5 || condition6);
+		return ((condition1 && condition2 && !condition3) || (condition3 && condition4)) && (condition5 || condition6);
```
Comment (abecerra) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java:242]: Why is negated now?

**Review by abecerra — APPROVED** [review]

**Comment by oblodgett** [comment]
@markquintontulloch can we get the other PR in first?

File: src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java (hunk 1 above)
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java:242]: @abecerra its not negated its added to the condition.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java:242]: If we apply the IRI prefix filter we don’t want terms to load if they match the default namespace but don’t have the correct prefix. ie. Passing condition1 and condition2 is not sufficient, checkIRIPrefix must be null

**Comment by markquintontulloch** [comment]
> 

Sure. I’m at my kids’ sports day so can’t review the other PR at the moment.  Will be back at my desk in about an hour.

**Comment by oblodgett** [comment]
@markquintontulloch this will probably fail... you will need to switch out the imports.

**Comment by oblodgett** [comment]
@markquintontulloch you will need to revert back to the javax imports and you can get this in for tomorrow.

**Pull request merged**